stan/math/prim/fun/variance.hpp

```diff
@@ -13,21 +13,23 @@ namespace math {
  * Returns the sample variance (divide by length - 1) of the
  * coefficients in the specified standard vector.
  *
- * @tparam T type of elements in the vector
+ * @tparam StdVec A standard library vector that does not contain a var.
  * @param v specified vector
  * @return sample variance of vector
  * @throw <code>std::invalid_argument</code> if the vector has size zero
  */
-template <typename T>
-inline return_type_t<T> variance(const std::vector<T>& v) {
+template <typename StdVec, require_std_vector_t<StdVec>* = nullptr,
+          require_not_vt_var<StdVec>* = nullptr>
+inline value_type_t<StdVec> variance(StdVec&& v) {
```

Comment:
Would it not be simpler to just wrap the argument in `Eigen::Map` and call Eigen overload?

Reply:
Yes!